controlplane/kubeadm/internal/control_plane.go

```diff
@@ -31,17 +32,25 @@ import (
 // ControlPlane holds business logic around control planes.
 // It should never need to connect to a service, that responsibility lies outside of this struct.
 type ControlPlane struct {
-	KCP      *controlplanev1.KubeadmControlPlane
-	Cluster  *clusterv1.Cluster
-	Machines FilterableMachineCollection
+	KCP            *controlplanev1.KubeadmControlPlane
+	Cluster        *clusterv1.Cluster
+	Machines       FilterableMachineCollection
+	MachineConfigs map[string]*bootstrapv1.KubeadmConfig
+	InfraObjects   map[string]*unstructured.Unstructured
+
+	// ReconciliationTime is the time of the current reconciliation, and should be used for all "now" calculations
+	ReconciliationTime metav1.Time
```

Comment:
We've had bugs around this where we are calling `Now` multiple times in a reconciliation, resulting in possibly different results for methods depending on how much time has passed since the reconciliation began. Doing something like this is cleaner than having to make sure we call these time-sensitive methods only once.